exporter/datasetexporter/config.go

```diff
@@ -44,11 +86,11 @@ func (c *Config) Unmarshal(conf *confmap.Conf) error {
 		c.DatasetURL = os.Getenv("DATASET_URL")
 	}
 	if len(c.APIKey) == 0 {
-		c.APIKey = os.Getenv("DATASET_API_KEY")
+		c.APIKey = configopaque.String(os.Getenv("DATASET_API_KEY"))
```

Comment:
we should not get this through an environment variable. The Collector has [builtin functionality to do this](https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables), so people would do:

```
exporters:
  dataset:
    api_key: ${env:DATASET_API_KEY}
```

to do this. We should instead fail if `len(c.APIKey) == 0`.

Reply:
Fixed. Thanks for pointing me to the documentation. I have removed this check from the `Unmarshal` function.